### ISSUE TITLE
Bonus Points and Homework

### ISSUE THREAD

**Issue by povrazor** [ludumdare/ludumdare#954]
When it comes to ratings, playing and rating certain games is more preferable to others.

* Web games (no download required, instant gratification)
* Popular games (usually there's something cool about them)
* Games on less-popular platforms (Mac, Linux)

To help convince players to play less desirable games, we can award **bonus points** for choosing to play and rate certain games over others.

There are other cases too where points can be awarded:

* An extra point for games under 20 ratings (May be feedback based)
* A bonus point for being the first to rate/get upvoted feedback on a game

To avoid abuse, certain conditions may apply (TBD, but probably author and vetted users).

In addition, we could assign everyone **5 Homework Games**. Doing your homework is worth additional bonus points.

**Comment by povrazor** [comment]
Yesterday I changed (again) how the Smart algorithm works, but only because it wasn't correct before. Now the sum-total priority points available to your game are 50 ratings, and 50 feedback karma. This works nicely, but it brings in to question how bonus points should work.

One possibility is that bonus points could affect the upper limit on ratings. Either lower the upper limit from 50 to 30, or let it raise the limit. Feedback karma in general is tougher to accumulate, so lowering this limit shouldn't be necessary. But if the upper limit on feedback was 30, and each homework game unlocked 1-2 more, it would add a greater incentive to choosing those games over others.

**Comment by local-minimum** [comment]
I like homework and incentives but it could be worth asking people when they upload a game by a google-drive link or similar if they are sure they want to do that because people usually feel a bit annoyed with it and maybe they should look at the hosting-page about better ways to upload the game. 

It would be a tricky line to decide which links should trigger a warning though and we don't want to be telling everyone they need to upload to itch.io even if I love that site (now with the itch-app, download install is almost as easy as web-game).

Also such warnings could trigger on using not as common compression-formats like `.rar`.

**Comment by ajayyy** [comment]
I don't get why people complain about drive vs itch.io though. Both require many clicks compared to hosting it somewhere like dropbox or github pages.

And I don't think it's a good idea to recommend itch.io because it is blocked in my school, making it so I can't rate those games there :)

**Comment by povrazor** [comment]
I wasn't really aware there was a criticism of Google drive.

**Comment by zwrawr** [comment]
There isn't that much, I think it come about purely because itch is tailored to games where as gDrive isn't.

**Comment by Cerno-b** [comment]
Personally I prefer games on itch because I use the app a lot. You don't need to unzip the game, as the adpp does it for you.

It would be even cooler if you could add a game to the app directly from the LD page, but I assume that wouldn't work (cross-site login etc). 

It's also much better for keeping games after LD (I only keep the itch ones) as you get a nice collection with title images etc.

Back on topic: I like the 5 homework games, more would be too much I think. I always keep a list of games to play and between cool games, and games from people who commented on mine, and rescue games, too many homework games would feel like a hassle, even if they are optional.

**Comment by Jezzamonn** [comment]
Another idea is to give a coolness boost that's inversely proportional to how many votes a game has. If you rate the all-star Ludum Darer's game which is probably going to already have 80 ratings, you get less of a boost than if you rate a game with 5 ratings.

There are lots of games that get low votes because they require weird platforms or extra set up (VR, multiplayer, old-school hardware), so we can't just randomly assign those. It'd be cool if there was more encouragement to play them.

**Comment by Cerno-b** [comment]
@Jezzamonn I like the idea of awarding coolness in proportion the the votes. That would probably work like the coolness drop-off: you need more outgoing votes per incoming votes the more votes you get. Still it would reward people voting early since then everyone has low votes. Could be tricky to optimize.

The platforms could be a problem though, not only for weird setups but also for normal situations (Linux, Windows, MacOS). The way around this would be to have checkboxes for the games (as is already planned) and checkboxes for what you can play (to be determined when you hand in your game).

**Comment by local-minimum** [comment]
My reason for personal strong `itch` preference for installed games compared other download alternatives is that I don't have to say where I install it, I can easily uninstall it afterwards and to some extent itch offers a boxing-thingy that may make it slightly more secure to install and run the games. Note I'm not talking about downloading through the site `itch.io` but using the `itch` app.

Many times people just place their files in a zip and not in a folder in the zip and that makes it some more labor doing it manually. It's not a big thing, but I do personally skip such downloads every other time because of the extra annoyance. I doubt I'm alone. I also find the dropbox and google drive pages confusing more often than not in that you may need to hunt around for where you actually download the full zip. On dropbox they have a `?dl=0` at the end of their uri and if you change it to `?dl=1` you at least don't need to see their page and download starts directly. (That at least we could suggest).

**Comment by ajayyy** [comment]
Ya, it does suck that Dropbox doesn't have direct downloads from the public folder anymore (there must be so many broken links in ludum dare pages now.

**Comment by TheMSB** [comment]
I'd like to chip in to this conversation by saying that I think useful feedback might actually be worth more than an arbitrary score. Having adding a slightly more complicated scoring system for feedback posts you could award people who take the time to give detailed feedback vs those who just leave a score.

**Comment by povrazor** [comment]
Yep. That's already a thing. That said, the only way we know a comment is good is if other people say it's good though.

**Comment by ajayyy** [comment]
I think using legnth of feedback could also help too, like how feedback friends does it.

**Comment by sgstair** [comment]
Re: Homework - I think this concept is immensely useful, and I suggest having ~10 or 20 random games assigned to you at voting start, where you get something like +0.5 smart rating boost per game of those you both rate + comment (30 words or more) on, or something like that.

**Comment by povrazor** [comment]
An API for this is roughed in (`/vx/homework.php`). It currently only supports Theme Suggestion mode, and includes a number of stubs for other things (submit a warmup game).

**Comment by bbi-danielsavage** [comment]
Has there been much headway on this feature? I think this would be a great fit for Ludum Dare's community-oriented aspect.

**Comment by mikekasprzak** [comment]
I have a branch called homework in my personal repo that I believe has changes that arent live yet. I can't remember what stalled it, but I have started going through my stalled branches (finished one this morning).

**Comment by Samusoidal** [comment]
Re: Google Drive and undesirable hosting platforms - 
Is there a tag or something for GDrive and other hosting services that allows an instant download? Kind of like tacking ?dl=1 to the end of a Dropbox link. If so, can we parse GDrive, Dropbox, etc links and tack that onto it to streamline download process?

**Comment by occultsoftworks** [comment]
I love this idea. So long as Homework is non-mandatory, I think it's a fantastic addition to the community's tool-set and rewards users for going above and beyond.

**Comment by Josh6680** [comment]
@Samusoidal For a normal GDrive link like this:
https://drive.google.com/open?id=1-d7HMqH2bNkpB5wE-U1GV6Cnjn3Q-isp
You can turn it into this for instant download:
https://drive.google.com/uc?export=download&id=1-d7HMqH2bNkpB5wE-U1GV6Cnjn3Q-isp

**Comment by Samusoidal** [comment]
@Josh6680 That's exactly what I was looking for, thanks!  

Some kind of parsing mechanism should be added to the games page link category to automatically add these. Not having the file shared correctly shouldn't be much of a problem because they have to have those settings for people to download them anyway. I've opened a separate issue on this.
#1656
